code/controllers/ModelAsController.php

```diff
@@ -49,82 +49,79 @@ public function init() {
 	public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 		$this->setRequest($request);
 		$this->setDataModel($model);
-		
-		$this->pushCurrent();
 
-		// Create a response just in case init() decides to redirect
-		$this->response = new SS_HTTPResponse();
```

Comment:
I've added the call without a return parameter to initialise the response on the request and have added a fix for the regression of the framework at https://github.com/silverstripe/silverstripe-framework/pull/4629